request_recorder.go

```diff
@@ -54,8 +66,14 @@ func (r *requestRecorder) String() string {
 		}
 		return str
 	}
-	accepted := requests2str(r.acceptedRequests)
-	unmatched := requests2str(r.unmatchedRequests)
+
+	r.mtx.RLock()
+	acceptedRequests := r.acceptedRequests
+	unmatchedRequests := r.unmatchedRequests
+	r.mtx.RUnlock()
+
+	accepted := requests2str(acceptedRequests)
```

Comment:
This should be done while the mutex is still locked because the slice might still change - new requests might be added the local variables are only referencing the same slices.

Reply:
Thanks! Fixed by extending mutex lock until end of the method.